### ISSUE TITLE
calcNhoodDistance doesn't honor requested `reduced.dim`

### ISSUE THREAD

**Issue by lianos** [MarioniLab/miloR#190]
Sorry if I'm misinterpreting the intention of the `reduced.dim` parameter in the `calcNhoodDistance()` function, but let's say I have two different "flavors" of a PCA reduction in my SingleCellExperiment ... something like this:

```r
reducedDimNames(sce)
[1] "PCA"  "PCA2"  "UMAP"  "UMAP2"
```

and I have been running the milo pipepline like so:

```r
milo <- Milo(sce)
milo <- buildGraph(milo, k = 20, d = 30, reduced.dim = "PCA2")
milo <- makeNhoods(milo, k = 20, d = 30, refined = TRUE, prop = 0.2,
                   reduced_dims = "PCA2")
```

Wouldn't I then want to call `calcNhoodDistance()` like so?

```r
milo <- calcNhoodDistance(milo, d = 30, reduced.dim = "PCA2")
```

However it looks like the code in the `calcNhoodDistance()` function will ignore the value of the `reduced.dim` parameter if it finds a reducedDim that is named `"PCA"`:

https://github.com/MarioniLab/miloR/blob/abeed5abb64b227ef2f0c95a9c4931b043c64a3f/R/calcNhoodDistance.R#L63-L74

but don't we want the `calcNhoodDistance()` to use `"PCA2"` in this case?

**Comment by MikeDMorgan** [comment]
Hi @lianos Thanks for raising this - I will patch this ASAP onto the devel branch and update when it's ready. Great catch.

**Comment by MikeDMorgan** [comment]
I've not patched this on the devel branch in 6d30ebdf87867f734468c9d28ffa1d2cfdf7bd71 - you should be able to install with `devtools::install_github("MarioniLab/miloR", ref="devel")`.

**Comment by lianos** [comment]
Great, thanks Mike!
